waspc/waspls/src/Wasp/LSP/Commands/Command.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE DataKinds #-}
+
+module Wasp.LSP.Commands.Command
+  ( Command (Command, commandName, commandHandler),
+    withParsedArgs,
+    makeInvalidParamsError,
+  )
+where
+
+import Control.Lens ((^.))
+import Data.Aeson (FromJSON, Result (Error, Success), Value, fromJSON)
+import Data.Text (Text)
+import qualified Data.Text as Text
+import qualified Language.LSP.Server as LSP
+import qualified Language.LSP.Types as LSP
+import qualified Language.LSP.Types.Lens as LSP
+import Wasp.LSP.ServerMonads (ServerM)
+
+-- | Command name and handler. When a 'LSP.WorkspaceExecuteCommand' request is
+-- received with the command name matching the one listed in a 'Command', the
+-- corresponding handler is executed.
+data Command = Command
+  { commandName :: Text,
+    commandHandler :: LSP.Handler ServerM 'LSP.WorkspaceExecuteCommand
+  }
+
+-- | @withParsedArgs request respond run@ parses args from a 'LSP.WorkspaceExecuteCommand'
+-- request and passes them to @run@. If an error occurs during parsing, responds
+-- with an error and does not execute @run@.
+withParsedArgs ::
+  (FromJSON args, LSP.MonadLsp c m) =>
+  -- | LSP 'request'.
+  LSP.RequestMessage 'LSP.WorkspaceExecuteCommand ->
+  -- | LSP 'respond'.
+  (Either LSP.ResponseError Value -> m ()) ->
+  -- | Handler that need arguments.
+  (args -> m ()) ->
+  m ()
+withParsedArgs request respond run = case request ^. LSP.params . LSP.arguments of
+  Just (LSP.List [argument]) -> case fromJSON argument of
+    Error err -> respond $ Left $ makeInvalidParamsError $ Text.pack err
+    Success args -> run args
+  _ -> respond $ Left $ makeInvalidParamsError "Expected exactly one argument"
```

Comment:
Took a pass at renaming variables and improving the comment, let me know if it's clearer!

Not sure on a more specific name: it is intended to be a pretty general function, so I don't know how much more specific it can be.

Reply:
Ok, this is certainly better!
I would probably try to improve the name still, I think it is just not clear from it what it does, and it could be. Ideas: `runCmdForRequest`, `parseLspRequestArgsAndRunCmd`.
But I will stop here, if you still think it is better to keep current name, ok, I do believe you have good reasons.